conversions/eastern-to-integer/eastern_to_integer.go

```diff
@@ -0,0 +1,26 @@
+/*
+This algorithm will convert an Eastern string number to an integer
+https://en.wikipedia.org/wiki/Eastern_Arabic_numerals
+Function receives a string as an Eastern number and outputs an integer
+*/
+
+package easterntointeger
+
+var persianNumbers = map[string]int{"۰": 0, "۱": 1, "۲": 2, "۳": 3, "۴": 4, "۵": 5, "۶": 6, "۷": 7, "۸": 8, "۹": 9}
+var arabicNumbers = map[string]int{"٠": 0, "١": 1, "٢": 2, "٣": 3, "٤": 4, "٥": 5, "٦": 6, "٧": 7, "٨": 8, "٩": 9}
+
+
+func EasternToInteger(stringNumber string) int {
+	if stringNumber == "" {
+		return 0
+	}
+	number := 0
+	for _, char := range stringNumber {
+		if value, ok := persianNumbers[string(char)]; ok { // Persian check
+			number = (number * 10) + value
+		} else if value, ok := arabicNumbers[string(char)]; ok { // Arabic check
+			number = (number * 10) + value
+		}
```

Comment:
What do you mean by "characters/numbers from any other language"? Can you give me an example?

Reply:
What happens when I pass a string "123"??

Reply:
Also consider what will happen when the input is "12۰34" where "۰" is the Persian sifr. Think about it

Reply:
I'll add errors for invalid characters/numbers. And in the case of inputs like "12۰34", I think using the single "۰" and ignoring the rest isn't a good idea. Validating such inputs requires more code and I think this time it will cross the line between best practice and simple enough so that the algorithm won't be lost between all the validations and cleanings 😄

Reply:
I didn't give that example as something that should work but rather I gave it so you can think about what the result of that would be😄 Of course it will get more convoluted if you do that but I don't think the string "12۰34" should even be a valid string for this function